File: API/Entities/ProductNutrition.cs
namespace NutriTrack.Entities;

public class ProductNutrition
{
    public int Id { get; set; }
    public int ProductNutritionCategoryId { get; set; }
    public string Name { get; set; }
    public decimal CaloriesPer100Grams { get; set; }
    public decimal ProteinPer100Grams { get; set; }
    public decimal FatPer100Grams { get; set; }
    public decimal CarbohydratesPer100Grams { get; set; }

    public string? ImageUrl { get; set; }

    public ProductNutritionCategory ProductNutritionCategory { get; set; }

    public List<ProductRecord> ProductRecords { get; set; }
}

File: client/src/app/models/productNutrition.ts
export interface ProductNutrition {
    id: number,
    name: string,
    caloriesPer100Grams: number,
    proteinPer100Grams: number,
    fatPer100Grams: number,
    carbohydratesPer100Grams: number,
    imageId?: string
}

export interface ProductNutritionParams {
    orderBy: string;
    searchTerm?: string;
    categories: ProductNutritionCategory[];
    pageNumber: number;
    pageSize: number;
}

export interface ProductNutritionCategory{
    id: number;
    name: string
    isDeleteable: boolean;
}

export default interface ProductRecordNew{
    productNutritionId: number;
    consumedGrams: number;
    date?: string
}